### PR TITLE
add Get New and Updated Objects Polling Trigger

### PR DESCRIPTION
**Pull request by shulkaolka**

### CODE REVIEW

File: component.json [hunk 1]
```diff
@@ -23,6 +23,57 @@
...
+        "pollConfig": {
+          "label": "Time stamp field to poll on",
+          "viewClass": "SelectView",
+          "required": false,
+          "order": 1,
+          "model": {
+            "created": "Created",
+            "lastModified": "Last Modified"
+          },
+          "prompt": "Please select a Time stamp field to poll on. Default: Time Updated"
```
Comment (EmptyInfinity) [inline component.json:72]:
```suggestion
          "prompt": "Please select a Time stamp field to poll on. Default: Last Modified"
```

File: component.json [hunk 2]
```diff
@@ -23,6 +23,57 @@
...
+            "todos": "Todo",
+            "users": "User"
+          }
+        },
+        "startTime": {
+          "label": "Start Time",
+          "viewClass": "TextFieldView",
+          "order": 5,
+          "placeholder": "1970-01-01T00:00:00.000Z",
+          "note": "Beginning time to start polling from (inclusive) in format \"YYYY-MM-DD[T]HH:MM:SS[Z]\" where [] wraps a fixed character value. Defaults to the beginning of time"
```
Comment (EmptyInfinity) [inline component.json:54]:
```suggestion
          "note": "Beginning time to start polling from (inclusive) in format \"YYYY-MM-DD[T]HH:MM:SS[Z]\" where [] wraps a fixed character value. Defaults to 1970-01-01T00:00:00.000Z""
```

File: component.json [hunk 3]
```diff
@@ -23,6 +23,57 @@
...
+      "main": "./lib/triggers/getNewAndUpdatedObjectsPolling.js",
+      "title": "Get New and Updated Objects Polling",
+      "help": {
+        "description": "Poll for new and/or updated objects"
+      },
+      "fields": {
+        "objectType": {
+          "label": "Object Type",
+          "viewClass": "SelectView",
+          "prompt": "Please select the type of object to poll for",
```
Comment (EmptyInfinity) [inline component.json:37]:
```suggestion
          "prompt": "Please select the type of object to poll on",
```

File: spec-integration/triggers/getNewAndUpdatedObjectsPolling.spec.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  it('should successfully retrieve a list of objects within a given start and end time and emit a snapshot', async () => {
+    cfg.startTime = '2019-03-17T16:33:08.809Z';
+    cfg.endTime = '2019-06-11T05:09:22.308Z';
+
+    await processTrigger.call(emitter, {}, cfg);
+    expect(emitter.emit.calledThrice).to.be.equal(true);
+    expect(emitter.emit.firstCall.args[1].body).to.deep.equal({
+      userId: 5,
+      id: 50,
+      title: 'repellendus qui recusandae incidunt voluptates tenetur qui omnis exercitationem',
```
Comment (EmptyInfinity) [inline spec-integration/triggers/getNewAndUpdatedObjectsPolling.spec.ts:31]: pls use English only

File: src/triggers/getNewAndUpdatedObjectsPolling.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import { messages } from 'elasticio-node';
+import Client from '../client';
+
+/**
```
Comment (EmptyInfinity) [inline src/triggers/getNewAndUpdatedObjectsPolling.ts:4]: pls remove comments

File: README.md [hunk 6]
```diff
@@ -3,13 +3,36 @@
...
 
+## Triggers
+
+### Get New and Updated Objects Polling
+
+Retrieve all the updated or created objects within a given time range.
+
+#### Configuration Fields
+
+- Object Type (required): One of the six object types available in the database
```
Comment (EmptyInfinity) [inline README.md:23]:
```suggestion
* **Object Type** (required): One of the six object types available in the database
```

Comment (EmptyInfinity) [reply README.md:23]: The same to next fields

File: src/triggers/getNewAndUpdatedObjectsPolling.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import { messages } from 'elasticio-node';
+import Client from '../client';
+
+export async function processTrigger(msg, cfg, snapshot = {
```
Comment (EmptyInfinity) [inline src/triggers/getNewAndUpdatedObjectsPolling.ts:4]: seems like `snaphot = { }` will give us same result

**Review by EmptyInfinity — APPROVED** [review]

File: src/triggers/getNewAndUpdatedObjectsPolling.ts (hunk 7 above)
Comment (shulkaolka) [reply src/triggers/getNewAndUpdatedObjectsPolling.ts:4]: No, tslint suggested this

**Pull request merged**